### PR TITLE
[docs] Fix Feature comparison 404 links

### LINKED ISSUE #1462: [docs] Data grid Feature comparison is outdated
Hi, this question is pretty business specific, so I'm not sure if it should go to the Stackoverflow, here or through some other channel. Sorry in advance.

I'm evaluating XGrid for the project I'm working on, and I believe the feature comparison is not up to date.
https://material-ui.com/components/data-grid/getting-started/#feature-comparison

For example, there's documentation for the Cell editing, which should mean that it already exists, however the table where the features are compared shows that it is still in the works.

Is the table not up to date, or is my assumption incorrect?

Also, some of the links are broken. For example [Components](https://material-ui.com/components/data-grid/rendering/#components).

### PR DESCRIPTION
**Pull request by ZeeshanTamboli**
Fixes #1462 

Preview: https://deploy-preview-1525--material-ui-x.netlify.app/components/data-grid/getting-started/#feature-comparison

### CODE REVIEW

**Comment by ZeeshanTamboli** [comment]
Perhaps we can include this in `v4.0.0-alpha.27` itself?

File: docs/src/pages/components/data-grid/getting-started/getting-started.md [hunk 1]
```diff
@@ -126,52 +126,52 @@ If you have an enterprise grid running with an expired or missing license key, t
...
+| [Excel export](/components/data-grid/export/#excel-export)                                |    ❌     |              ❌               |                  🚧                   |
+| [Clipboard](/components/data-grid/export/#clipboard)                                      |    ❌     |              🚧               |                  🚧                   |
+| **Rendering**                                                                             |           |                               |                                       |
+| [Column virtualization](/components/data-grid/virtualization/#column-virtualization)      |    ✅     |              ✅               |                  ✅                   |
+| [Row virtualization > 100 rows](/components/data-grid/virtualization/#row-virtualization) |    ❌     |              ✅               |                  ✅                   |
+| [Components](/components/data-grid/components/)                                           |    ✅     |              ✅               |                  ✅                   |
```
Comment (oliviertassinari) [inline docs/src/pages/components/data-grid/getting-started/getting-started.md:164]: "Components" as a title doesn't make sense. I don't know why I used this. How about?

```suggestion
| [Customization](/components/data-grid/components/)                                           |    ✅     |              ✅               |                  ✅                   |
```

**Comment by oliviertassinari** [comment]
@ZeeshanTamboli Release on npm and of the docs are not completely correlated. We could make a release on npm and fix the docs even after.

File: docs/src/pages/components/data-grid/getting-started/getting-started.md (hunk 1 above)
Comment (ZeeshanTamboli) [reply docs/src/pages/components/data-grid/getting-started/getting-started.md:164]: The link is to `/components/`. Maybe:
```suggestion
| [Customizable components](/components/data-grid/components/)                                           |    ✅     |              ✅               |                  ✅                   |
```

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/getting-started/getting-started.md:164]: Ok

**Comment by oliviertassinari** [comment]
@ZeeshanTamboli Thanks

**Pull request merged**